### PR TITLE
OOP examples

### PR DESCRIPTION
**Pull request by takekoputa**
- [x] Class
- [x] Class variables / Object variables
- [x] Abstract classes / Abstract functions

### CODE REVIEW

**Comment by takekoputa** [comment]
This is ready for review

File: materials/introduction/02-gem5-basics/02-frequently-used-features/02-class-variables.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+        self.freq = freq
+
+    def change_type_name(self, new_name):
+        self.type_name = new_name  # this does not change Processor.type_name
+        # it creates an object variable instead
+
+
+class ProcessorWithClassFunction(Processor):
+    def __init__(self, freq):
+        super(ProcessorWithClassFunction, self).__init__(freq)
```
Comment (powerjg) [inline materials/introduction/02-gem5-basics/02-frequently-used-features/02-class-variables.py:14]: I suggest using simply `super()` as it should work unless there's weird multiple inheritance.

File: materials/introduction/02-gem5-basics/02-frequently-used-features/02-class-variables.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
+class Processor:
+    type_name = "Processor"
```
Comment (powerjg) [inline materials/introduction/02-gem5-basics/02-frequently-used-features/02-class-variables.py:2]: I'd use something other than the type name as that could be confusing. Maybe `default_frequency = "3GHz"`?

File: materials/introduction/02-gem5-basics/02-frequently-used-features/02-class-variables.py (hunk 1 above)
Comment (takekoputa) [reply materials/introduction/02-gem5-basics/02-frequently-used-features/02-class-variables.py]: Done

**Pull request merged**